### PR TITLE
CORDA-1670 fix multiple CorDapps for flow check

### PR DESCRIPTION
**Pull request by tudor-malene**
this is causing: https://r3-cev.atlassian.net/browse/CORDA-1670

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/internal/cordapp/CordappLoader.kt [hunk 1]
```diff
@@ -50,7 +50,7 @@ class CordappLoader private constructor(private val cordappJarPaths: List<Restri
         cordapps.flatMap { corDapp -> corDapp.allFlows.map { flow -> flow to corDapp } }
                 .groupBy { it.first }
                 .mapValues {
-                    require(it.value.size == 1) { "There are multiple CorDapp jars on the classpath for flow ${it.value.first().first.name}: ${it.value.map { it.second.name }.joinToString()}." }
+                    if(it.value.size > 1) { throw MultipleCordappsForFlowException("There are multiple CorDapp JARs on the classpath for flow ${it.value.first().first.name}: [ ${it.value.map { it.second.name }.joinToString()} ].")  }
```
Comment (tlil) [inline node/src/main/kotlin/net/corda/node/internal/cordapp/CordappLoader.kt]: `if (` missing space

`map { ... }.joinToString()` can be rewritten to `joinToString { ... }` :)

File: node/src/main/kotlin/net/corda/node/internal/cordapp/CordappProviderImpl.kt [hunk 2]
```diff
@@ -41,6 +41,9 @@ open class CordappProviderImpl(private val cordappLoader: CordappLoader,
 
     private fun verifyInstalledCordapps(attachmentStorage: AttachmentStorage) {
 
+        // This will invoke the lazy flowCordappMap property, thus triggering the MultipleCordappsForFlow check.
+        cordappLoader.flowCordappMap
```
Comment (tlil) [inline node/src/main/kotlin/net/corda/node/internal/cordapp/CordappProviderImpl.kt]: And we're guaranteed this is never done prematurely, right?

Comment (tudor-malene) [reply node/src/main/kotlin/net/corda/node/internal/cordapp/CordappProviderImpl.kt]: Yes, it's run after the cordapps were loaded.
Before this PR it was only run if there were checkpoints in the db.

Comment (tlil) [reply node/src/main/kotlin/net/corda/node/internal/cordapp/CordappProviderImpl.kt]: Gotcha

**Pull request merged**